### PR TITLE
Fix #89: Neutron: add time fields [network/subnet/port]

### LINKED ISSUE #89: I hope to get created_time and updated_time field but can not
When creating a network or subnet, I want to get the created time field and updated time field, but openstack4j does not provide it.

### PR DESCRIPTION
**Pull request by bboyHan**
#89 fixed the issue

### CODE REVIEW

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.openstack4j.model.common;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (olivergondza) [inline core/src/main/java/org/openstack4j/model/common/TimeEntity.java:3]: Unused import

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * @author bboyHan
+ */
+public interface TimeEntity {
+
+    /**
+     * created time
+     * 
+     * @return created time
+     */
+    String getCreatedTime();
```
Comment (olivergondza) [inline core/src/main/java/org/openstack4j/model/common/TimeEntity.java:17]: Do not return time as `String`. Rest of the codebase is using `Date`.

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java (hunk 1 above)
Comment (olivergondza) [inline core/src/main/java/org/openstack4j/model/common/TimeEntity.java:3]: Unused import

**Review by olivergondza — CHANGES_REQUESTED** [review]
Please add unittests verifying this in https://github.com/openstack4j/openstack4j/blob/master/core-test/src/main/java/org/openstack4j/api/network/NetworkTests.java

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java (hunk 1 above)
Comment (bboyHan) [reply core/src/main/java/org/openstack4j/model/common/TimeEntity.java]: got it

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java (hunk 2 above)
Comment (bboyHan) [reply core/src/main/java/org/openstack4j/model/common/TimeEntity.java]: got it

File: core/src/main/java/org/openstack4j/model/common/TimeEntity.java (hunk 1 above)
Comment (bboyHan) [reply core/src/main/java/org/openstack4j/model/common/TimeEntity.java]: got it

File: core-test/src/main/java/org/openstack4j/api/AbstractTest.java [hunk 3]
```diff
@@ -71,6 +72,7 @@ private Service(int port) {
     protected static final String JSON_ACCESS = "/identity/v2/access.json";
     protected static final String JSON_TOKEN = "/identity/v3/authv3_project.json";
     protected static final String TOKEN_ID = "123456789";
+    protected static final Date DATE = new Date(1604096161000L);
```
Comment (olivergondza) [inline core-test/src/main/java/org/openstack4j/api/AbstractTest.java:75]: Do not pull this into `AbstractTest`. It is irrelevant for most of the implementations.

File: core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronNetwork.java [hunk 4]
```diff
@@ -52,9 +51,11 @@
     @JsonProperty("availability_zones")
     private List<String> availabilityZones;
     @JsonProperty("created_at")
-    private String createdTime;
+    @JsonFormat(pattern="yyyy-MM-dd'T'HH:mm:ss")
```
Comment (olivergondza) [inline core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronNetwork.java:54]: Is specifying the format really needed? A lot of date fields is not having those at all.

Comment (bboyHan) [reply core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronNetwork.java]: This is not really a format that must be specified.

Comment (bboyHan) [reply core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronNetwork.java]: Thank you for your suggestion. I haven't considered it comprehensively.

**Comment by olivergondza** [comment]
Thanks for your contribution, @bboyHan. I am merging this towards next release.

**Pull request merged**